### PR TITLE
Add helper methods on `Visibility`

### LINKED ISSUE #14825: Make toggling visibility easier
## What problem does this solve or what need does it fill?

Quickly toggling visibility is a common operation. We should make it even easier.

## What solution would you like?

Implement `Not` for `Visibility`. Document the behavior of `Visibility::Inherited` (left unchanged) in the doc comment for the impl block.

## What alternative(s) have you considered?

Convert to and from a bool using https://github.com/bevyengine/bevy/pull/14784. This is likely slower, and much more verbose.

Similarly, writing your own match statement is tedious and verbose.

We could also (or instead) add a `.toggle` method, and have one variant which converts Inherited -> Hidden, and one which leaves it alone.

### PR DESCRIPTION
**Pull request by ThomasAlban**
Fixes #14825

Edit: After feedback, these are the updated methods:

- `toggle_inherited_visible(&mut self)`
    - Toggles between `Visibility::Inherited` and `Visibility::Visible`. If the value is `Visibility::Hidden`, it remains unaffected.
- `toggle_inherited_hidden(&mut self)`
    - Toggles between `Visibility::Inherited` and `Visibility::Hidden`. If the value is `Visibility::Visible`, it remains unaffected.
- `toggle_visible_hidden(&mut self)`
    - Toggles between `Visibility::Visible` and `Visibility::Hidden`. If the value is `Visibility::Inherited`, it remains unaffected.

### CODE REVIEW

**Comment by ThomasAlban** [comment]
I don't know why my commits from a previous PR which was merged are also included here, but they don't affect the files changed as it was merged anyway 🙄 github fail

**Comment by ickshonpe** [comment]
> I don't know why my commits from a previous PR which was merged are also included here, but they don't affect the files changed as it was merged anyway 🙄 github fail

yeah happens to me all the time 😅

File: crates/bevy_render/src/view/visibility/mod.rs [hunk 1]
```diff
@@ -47,6 +47,55 @@ pub enum Visibility {
...
+    pub fn from_bool_inherited(value: bool) -> Self {
+        if value {
+            Visibility::Inherited
+        } else {
+            Visibility::Hidden
+        }
+    }
+    /// Returns `true` if `self` is `Visibility::Visible` or `Visibility::Inherited`, else returns `false`.
+    #[inline]
+    pub fn to_bool_inherited(self) -> bool {
```
Comment (ickshonpe) [inline crates/bevy_render/src/view/visibility/mod.rs:62]: I'm not sure about the naming scheme. `to_bool_inherited` doesn't feel very intuitive, `visibility != Visibility::Hidden` seems much clearer.

File: crates/bevy_render/src/view/visibility/mod.rs [hunk 2]
```diff
@@ -47,6 +47,55 @@ pub enum Visibility {
...
+    pub fn from_bool_unconditional(value: bool) -> Self {
+        if value {
+            Visibility::Visible
+        } else {
+            Visibility::Hidden
+        }
+    }
+    /// Returns `true` if `self` is `Visibility::Visible`, else returns `false`.
+    #[inline]
+    pub fn to_bool_unconditional(self) -> bool {
```
Comment (ickshonpe) [inline crates/bevy_render/src/view/visibility/mod.rs:85]: I don't think i'd ever be able to remember with confidence what `to_bool_unconditional` did without looking at the docs.
```pub fn is_unconditionally_visible(self) -> bool {``` would be better maybe? 
But neither seems as readable `visibility == Visibility::Visible` really.

**Review by ickshonpe — COMMENTED** [review]
`to_bool_` and `from_bool_` don't seem very idiomatic, `is_` and `if_` would make more sense I think.

**Comment by ThomasAlban** [comment]
Converted to draft as I feel like there is room for improvement here

**Comment by janhohenheim** [comment]
Imo these methods do not have clear naming for an outsider and are nonetheless on the verge of being verbose enough that I would probably personally prefer a manual `matches!`.

**Comment by ThomasAlban** [comment]
from discussion on bevy discord, how about this instead:
```rust
toggle_inherited_visible()
toggle_inherited_hidden()
toggle_visible_hidden()
```
these methods toggle between the 2 states in the name

**Comment by ThomasAlban** [comment]
Yeah this problem is quite hard to solve but I'm determined to find a solution which covers at least some common use cases. I initially thought my wording was quite clear but from what you guys have said it seems to need work! Let me know what you think of the above suggestion (suggested by bestRanar on bevy discord)

**Comment by janhohenheim** [comment]
@ThomasAlban assuming that these functions simply ignore the variant that is not being toggled, I think those are good :) I can see myself using them!
Btw, that would be @rparrett on GitHub

**Comment by ThomasAlban** [comment]
The Perhaps to bool methods should be removed in favour of simply doing a `matches!` statement or something, but Methods to convert from bool to visibility in some way would be particularly useful to me in my codebase. Does anyone have any alternate ideas on naming for these methods?

**Comment by janhohenheim** [comment]
@ThomasAlban agreed that those two would be the most useful ones. Maybe `Visibility::inherited_if` and `Visibility::visible_if`?

**Review by janhohenheim — APPROVED** [review]
I think this is alright now. Good job gradually improving the PR :)

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
I like the toggle methods, but the `if` family of methods seems both very niche and challenging to read. If you cut those from here I'll approve and merge this.

**Comment by ThomasAlban** [comment]
@alice-i-cecile In my codebase there are loads of places where I have to do
```rust
if value {
    Visibility::Visible
} else {
    Visibility::Hidden
}
```
And it would be so nice if it could be cut down to a single function call such as what I proposed.
Are there any changes that could be made to the naming etc that would improve them for you?
If it's a definite no, I'll remove them :)

**Comment by janhohenheim** [comment]
Can confirm that I have also seen that pattern crop up a few times. I personally don't use it much because I tend to make my `UiNode` explicitly visible or invisible through events, but I think this is a valid use-case.

**Comment by alice-i-cecile** [comment]
IMO the `if` pattern is much more readable, sorry!

**Review by alice-i-cecile — APPROVED** [review]
Great, thank you <3

**Pull request merged**